client/core/trade.go

```diff
@@ -1007,8 +1011,10 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) boo
 			coinIDString(wallet.AssetID, match.MetaData.Proof.MakerSwap), unbip(wallet.AssetID))
 		confs, spent, err := wallet.SwapConfirmations(ctx, match.MetaData.Proof.MakerSwap,
 			match.MetaData.Proof.ContractData, match.MetaData.Stamp)
-		if err != nil {
-			t.dc.log.Errorf("error getting confirmation for our own swap transaction: %v", err)
+		if err != nil && !errors.Is(err, asset.ErrSwapNotInitiated) {
+			// No need to log an error if swap not initiated as this
+			// is expected for newly made swaps involving contracts.
+			t.dc.log.Errorf("isSwappable: error getting confirmation for our own swap transaction: %v", err)
 		}
```

Comment:
This is closely related to #1425, and `ErrSwapNotInitiated` is analogous to `CoinNotFoundError`, as far as I can tell. There was [some discussion on this on Element too](https://app.element.io/#/room/!SFRQQFIHUUNXARfvew:decred.org/$RcebZlyaATD7J8ZtuI-PqAurABg2VFcdGTgwzBYKbGs). On that end, chappjc decided to just return `nil` error and zero confs in that case. Is there a strong argument to do it differently here?

Reply:
Just want to reiterate that https://github.com/decred/dcrdex/pull/1425 is just a stopgap for v0.4, and if we do keep it `nil` when the swap isn't found (either txn with utxo assets or an entry in the contracts map for eth), I suspect we'll want a `spent bool` or similar from `SwapConfirmations`, with an error reserved for the case where it is actually unexpected to not find it, but of course when unconfirmed it is expected (proper error semantics).

Reply:
meant to say **found** above, not spent, sorry

Reply:
I am trying to avoid returning a false positive, I guess. Not finding the swap could be a problem or it might not be. It's not that there is absolutely no problem, there may be a problem. Does that make sense? We could use the initiation tx here instead and confidently return `CoinNotFoundError` like the server does.

Reply:
Maybe it would be more correct to log as warn?

Reply:
It depends on (a) the capabilities of the wallet backend and (b) time and blocks that have elapsed.  I've beat this horse to death now, but I am adamantly opposed to logging a warning or error if it's normal for the wallet not to find the unconfirmed swap.

Reply:
No need to change this now, but I think the ultimate solution involves more work in Core to be intelligent about gauging severity.